sdk/servicebus/Azure.Messaging.ServiceBus/src/Management/ServiceBusManagementClient.cs

```diff
@@ -306,12 +362,21 @@ public virtual async Task<Response<NamespaceProperties>> GetNamespacePropertiesA
             CancellationToken cancellationToken = default)
         {
             EntityNameFormatter.CheckValidQueueName(name);
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(ServiceBusManagementClient)}.GetQueue");
+            scope.Start();
 
-            Response response = await _httpRequestAndResponse.GetEntityAsync(name, null, false, cancellationToken).ConfigureAwait(false);
-            var result = await ReadAsString(response).ConfigureAwait(false);
-            QueueProperties properties = QueuePropertiesExtensions.ParseFromContent(result);
-
-            return Response.FromValue(properties, response);
+            try
+            {
+                Response response = await _httpRequestAndResponse.GetEntityAsync(name, null, false, cancellationToken).ConfigureAwait(false);
+                var result = await ReadAsString(response).ConfigureAwait(false);
+                QueueProperties properties = QueuePropertiesExtensions.ParseFromContent(result);
+                return Response.FromValue(properties, response);
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
```

Comment:
I feel like I've mentioned this several months ago, but I forgot the answer. Since there is this pattern of `using DiagnosticScope scope` and the Start and the try/catch, isn't there some way to just make that a static method and provide the body of the try with a Func?

Reply:
Hmm, I'm not sure about that. This is the pattern that seems to be prevalent throughout the T2 libraries.

Reply:
I think it had to do with the fact the callsite for the scope needs to be in the method that the scope is for. So, if you extract that pattern to a separate method, it bricks the logic for using the scope correctly.